### ISSUE TITLE
Can't handle blank subjects

### ISSUE THREAD

**Issue by tpendragon** [ruby-rdf/rdf-raptor#23]
https://gist.github.com/tpendragon/1ad503fddf769567cf21

Also interesting,
```ruby
g.dump(:ntriples) # => ""
```

**Comment by tpendragon** [comment]
Wondering if we need to enable RAPTOR_FEATURE_RELATIVE_RELATIVE_URIs, but not sure how to set features in this C binding.

**Comment by dwbutler** [comment]
Looks like we would need to expose [raptor_serializer_set_option](http://librdf.org/raptor/api/raptor2-section-serializer.html#raptor-serializer-set-option) and [raptor_parser_set_option](http://librdf.org/raptor/api/raptor2-section-parser.html#raptor-parser-set-option).

**Comment by bendiken** [comment]
Exposing those two methods for setting options looks fairly straightforward based on their API signatures, other than the mechanical work involved in defining the big `raptor_option` enum.

I'd be a little surprised, though, if the `RAPTOR_FEATURE_RELATIVE_URIS` option enables @tpendragon's example to work. As I understand it, that feature is for Raptor (itself) to auto-abbreviate URIs when serializing into an output format that supports qualified names.

`raptor_uri` instances are supposed to be absolute URIs, which means that the expansion of relative URIs to absolute needs to be done on construction, you can't feed the serializer a statement with a relative URI:
* http://librdf.org/raptor/api/raptor2-section-uri.html#raptor-uri
* http://librdf.org/raptor/api/raptor2-section-uri.html#raptor-new-uri-relative-to-base

**Comment by tpendragon** [comment]
@bendiken Yeah, I tried it last night and it didn't work. Is this just unsupported by raptor?

**Comment by gkellogg** [comment]
> Exposing those two methods for setting options looks fairly straightforward based on their API signatures, other than the mechanical work involved in defining the big raptor_option enum.
> 
> I'd be a little surprised, though, if the RAPTOR_FEATURE_RELATIVE_URIS option enables @tpendragon's example to work. As I understand it, that feature is for Raptor (itself) to auto-abbreviate URIs when serializing into an output format that supports qualified names.
> 
This is the same as the standard_prefixes option with other serializers, so that would be a good way to expose it too.
> raptor_uri instances are supposed to be absolute URIs, which means that the expansion of relative URIs to absolute needs to be done on construction, you can't feed the serializer a statement with a relative URI:
> 
> http://librdf.org/raptor/api/raptor2-section-uri.html#raptor-uri
> http://librdf.org/raptor/api/raptor2-section-uri.html#raptor-new-uri-relative-to-base
That is generally true for RDF serializers, but some play loose with the rules :). Setting a base URI may help, which could be removed from the serialization afterwards if necessary.
